### PR TITLE
Fixing perf regression in marking TTI for workloads page

### PR DESCRIPTION
**Pull request by rakol-ms**
Containing `totalNodesRendered` in state causes an infinite trigger of updates for all the child components. Everytime a child components calls `markTTICallback`, count is updated which in turn causes child components to re-render triggering and marking TTI again and process gets stuck in an infinite loop. Causing slowed rendering of UI.

Fix:
1. Moved totalNodesRendered to a class variable.
2. Notifying the TTI callback from child components only after the data needed in child components are updated. This will track the entire load time of the scenario instead of time taken for the child components to mount for first time (without data loads returning).

### CODE REVIEW

File: src/WebUI/Workloads/WorkloadsPivot.tsx [hunk 1]
```diff
@@ -107,15 +104,18 @@ export class WorkloadsPivot extends React.Component<IWorkloadsPivotProps, IWorkl
...
+            });
         }
     }
 
-    private _notifyRender = () => {
-        // const nodeCount = this.state.totalNodesRendered;
-        // this.setState({
-        //     totalNodesRendered: nodeCount + 1
-        // });
+    private _notifyRender = (props?: { [key: string]: any }) => {
```
[the remark this answers is not in the document]
Comment (rakol-ms) [reply src/WebUI/Workloads/WorkloadsPivot.tsx]: Ideally we can pass `additionalProperties` to `scenarioEnd()`. I am not sure how I can utilize it here.

File: src/WebUI/Workloads/OtherWorkloadsTable.tsx [hunk 2]
```diff
@@ -124,14 +123,14 @@ export class OtherWorkloads extends React.Component<IOtherWorkloadsProperties, I
         const storeState = this._store.getState();
         this.setState({
             statefulSets: storeState.statefulSetList && storeState.statefulSetList.items || []
-        });
+        }, this._onMetadataFetch);
```
Comment (madhuv-msft) [inline src/WebUI/Workloads/OtherWorkloadsTable.tsx]: one idea i have in this case is, have a dictionary with 4 values, like a["daemon"]=false etc, then at each of the call set the value to true

in the final _onMetadataFetch, check for all keys of the dictionary and see value is true and then call the original one, so you do not have to rely on === 4

File: src/WebUI/Workloads/WorkloadsPivot.tsx [hunk 3]
```diff
@@ -107,15 +104,18 @@ export class WorkloadsPivot extends React.Component<IWorkloadsPivotProps, IWorkl
     private _onDataFound = (): void => {
         const workloadSize = this._workloadsStore.getWorkloadSize();
         if (this.state.workloadResourceSize <= 0 && workloadSize > 0) {
-            this.setState({ workloadResourceSize: workloadSize });
+            this.setState({ workloadResourceSize: workloadSize }, () => {
+                // resettting counts upon workloads details load
+                this._childNodes = 0;
```
Comment (madhuv-msft) [inline src/WebUI/Workloads/WorkloadsPivot.tsx]: here also we can follow dictionary mode, as callers are calling only once. so we can have deployment and other kind of key and make it true, based on the key that we receive in the notifyrender. then check for all keys, if dict[key] is true then we are good.

File: src/WebUI/Workloads/DeploymentsTable.tsx [hunk 4]
```diff
@@ -84,6 +81,8 @@ export class DeploymentsTable extends React.Component<IDeploymentsTablePropertie
         this.setState({
             deploymentList: storeState.deploymentList,
             replicaSetList: storeState.replicaSetList
+        }, () => {
+            this.props.markTTICallback && this.props.markTTICallback({ "component": "Deployment table"});
```
[the remark this answers is not in the document]
Comment (rakol-ms) [reply src/WebUI/Workloads/DeploymentsTable.tsx]: Using it now to wait in parent Component

File: src/WebUI/Workloads/WorkloadsPivot.tsx (hunk 3 above)
Comment (rakol-ms) [reply src/WebUI/Workloads/WorkloadsPivot.tsx]: Done. Used a dictionary to track the metadata loading in `OtherWorkloads` page and another dictonary to track child components to be in ready state in `WorkloadsPivot`

File: src/WebUI/Workloads/OtherWorkloadsTable.tsx (hunk 2 above)
Comment (rakol-ms) [reply src/WebUI/Workloads/OtherWorkloadsTable.tsx]: Done.

File: src/WebUI/Workloads/WorkloadsPivot.tsx [hunk 5]
```diff
@@ -107,15 +100,25 @@ export class WorkloadsPivot extends React.Component<IWorkloadsPivotProps, IWorkl
...
     }
 
-    private _notifyRender = () => {
-        // const nodeCount = this.state.totalNodesRendered;
-        // this.setState({
-        //     totalNodesRendered: nodeCount + 1
-        // });
+    private _notifyRender = (props?: { [key: string]: string }) => {
+        if (!this._isTTIMarked && props && this._componentsInitialized) {
+            this._componentsInitialized[props["component"]] = true;
```
Comment (madhuv-msft) [inline src/WebUI/Workloads/WorkloadsPivot.tsx]: props["component"] this could be null/undefined, so check and then assign

**Pull request merged**